src/pypi_mapping/custom_pypi_mapping.rs

```diff
@@ -161,27 +162,41 @@ fn amend_pypi_purls_for_record(
         return Ok(());
     }
 
+    let mut not_a_pypi = false;
+
     // If this package is a conda-forge package or user specified a custom channel mapping
     // we can try to guess the pypi name from the conda name
-    if custom_mapping.contains_key(&record.channel) {
-        if let Some(mapped_channel) = custom_mapping.get(&record.channel) {
-            if let Some(mapped_name) =
-                mapped_channel.get(record.package_record.name.as_normalized())
-            {
-                record.package_record.purls.push(
-                    PackageUrl::new(String::from("pypi"), mapped_name)
-                        .expect("valid pypi package url"),
-                );
+    if let Some(mapped_channel) = custom_mapping.get(&record.channel) {
+        if let Some(mapped_name) = mapped_channel.get(record.package_record.name.as_normalized()) {
+            if let Some(name) = mapped_name {
```

Comment:
It think you can use an if-let chain for this: https://github.com/rust-lang/rust/pull/88642. Not sure, have not used it yet.

Reply:
it is saying that ``let` expressions in this position are unstable`
https://github.com/rust-lang/rust/issues/53667